### PR TITLE
CON-1025-upgrade-commitments-client-to-use-azure-ad

### PR DESCRIPTION
**Pull request by Stephek**

### CODE REVIEW

File: src/SFA.DAS.Forecasting.Azure.Resources/deployments/function-app-resources.json [hunk 1]
```diff
@@ -116,14 +131,34 @@
               "name": "Commitments-ClientToken",
               "value": "[parameters('CommitmentsClientToken')]"
             },
+            {
+              "name": "Commitments-ApiBaseUrl",
```
Comment (adam230594) [inline src/SFA.DAS.Forecasting.Azure.Resources/deployments/function-app-resources.json]: Can the settings names all be renamed to the parameter name eg "CommitmentsV1ApiBaseUrl". It won't be consistent with the old JWT authentication settings but they could also be changed too.

File: src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/ConfigurationHelper.cs [hunk 2]
```diff
@@ -50,6 +50,11 @@ public static CommitmentsApiConfig GetCommitmentsApiConfiguration()
                 {
                     BaseUrl = CloudConfigurationManager.GetSetting("Commitments-BaseUrl"),
                     ClientToken = CloudConfigurationManager.GetSetting("Commitments-ClientToken"),
+                    ApiBaseUrl = CloudConfigurationManager.GetSetting("Commitments-ApiBaseUrl"),
```
Comment (adam230594) [inline src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/ConfigurationHelper.cs]: These will have to be changes as well to follow the  new app settings

File: src/SFA.DAS.Forecasting.Commitments.Functions/local.settings.json [hunk 3]
```diff
@@ -23,7 +23,12 @@
     "ApprenticeshipsApiBaseUri": "https://findapprenticeshiptraining-api.sfa.bis.gov.uk/",
     "APPINSIGHTS_INSTRUMENTATIONKEY": "f0ca1c02-2018-4445-b2e8-001e322bf0e8",
     "Commitments-BaseUrl": "http://localhost:7090/",
-    "Commitments-ClientToken": "Whatever I think"
+    "Commitments-ClientToken": "Whatever I think",
+    "Commitments-ApiBaseUrl": "",
```
Comment (adam230594) [inline src/SFA.DAS.Forecasting.Commitments.Functions/local.settings.json]: These too

File: src/SFA.DAS.Forecasting.Azure.Resources/deployments/function-app-resources.json (hunk 1 above)
Comment (Stephek) [reply src/SFA.DAS.Forecasting.Azure.Resources/deployments/function-app-resources.json]: Ok

File: src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/ConfigurationHelper.cs (hunk 2 above)
Comment (Stephek) [reply src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/ConfigurationHelper.cs]: Ok

File: src/SFA.DAS.Forecasting.Commitments.Functions/local.settings.json (hunk 3 above)
Comment (Stephek) [reply src/SFA.DAS.Forecasting.Commitments.Functions/local.settings.json]: Ok

**Pull request merged**